packages/frontend/src/lib/dashboard/Dashboard.svelte

```diff
@@ -52,36 +48,18 @@ async function pullExampleImage(): Promise<void> {
       // Ensure UI updates to reflect the new state
       tick().catch((e: unknown) => console.error('error updating disclaimer', e));
     }
-  }, 5000);
-}
+  }, 5_000);
 
-onMount(async () => {
-  bootcAvailableImages = await bootcClient.listBootcImages();
+  await bootcClient.pullImage(exampleImage);
```

Comment:
something more robust would to do the following
```suggestion
  await bootcClient.pullImage(exampleImage).finally(() => {
    pullInProgress = false;
    displayDisclaimer = false;
  });
```

this would ensure that if pullImage throw an error, we still stop the loading animation

Reply:
You can also use the non-promise version

```suggestion
  try {
    await bootcClient.pullImage(exampleImage);
  } finally {
    pullInProgress = false;
    displayDisclaimer = false;
  }
```

Reply:
You're right, thanks. 👍🏼 Went with the first option in new commit.